File: src/routes/subscription.routes.ts
import {Hono} from "hono";
import type {Env} from "@/types/hono";
import {authMiddleware} from "@/middleware/auth";
import {SubscriptionService} from "@/services/subscription.service";
import {successResponse} from "@/utils/response";

export const subscriptionRoutes = new Hono<Env>();
const subscriptionService = new SubscriptionService();

subscriptionRoutes.use("*", authMiddleware);

subscriptionRoutes.get("/info", async (c) => {
  const user = c.get('user');

  const info = await subscriptionService.getInfo(user);

  return c.json(successResponse(info), 200);
});

File: fisoft/src/main/kotlin/routes/SubscriptionRoutes.kt
package com.ilyasdemirkiran.routes

import com.ilyasdemirkiran.middleware.authenticateUser
import com.ilyasdemirkiran.middleware.currentCompanyId
import com.ilyasdemirkiran.services.SubscriptionService
import com.ilyasdemirkiran.utils.successResponse
import io.ktor.http.*
import io.ktor.server.application.*
import io.ktor.server.response.*
import io.ktor.server.routing.*

fun Route.subscriptionRoutes() {
    val subscriptionService = SubscriptionService()

    route("/api/subscription") {
        get {
            call.authenticateUser()
            val companyId = call.currentCompanyId
            val sub = subscriptionService.getSubscription(companyId)
            call.respond(HttpStatusCode.OK, successResponse(sub))
        }
    }
}
